client/src/components/bugsnag/__tests__/BugSnagUserHandler.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import BugSnagUserHandler from '../BugSnagUserHandler';
+
+test('lifecycle', async () => {
+  const bugsnag = {
```

Comment:
you could use jest mocks - might make it a bit simpler
```
const bugsnag = {
  setUser: jest.fn(),
}
```
I'd also suggest putting this outside the test because it can then be defined in one place and used by all tests. All tests run with independent module globals so you don't need to worry about conflicts.

Reply:
I put it outside the test and it seemed to not be independent, calls length didnt reset by the end of it